gcsio/src/main/java/com/google/cloud/hadoop/gcsio/GoogleCloudStorageImpl.java

```diff
@@ -2342,6 +2345,11 @@ <RequestT extends StorageRequest<?>> RequestT initializeRequest(
       String token = downscopedAccessTokenFn.apply(accessBoundaries);
       request.getRequestHeaders().setAuthorization("Bearer " + token);
     }
+    if (storageOptions.isGcsAuditLogEnabled()
+        && !storageOptions.getHttpRequestHeaders().containsKey(CUSTOM_AUDIT_USER_HEADER)) {
```

Comment:
why we need this check? Customer is supposed to set this?

Reply:
why we need an additional flag (GcsAuditLogEnabled) if having this header is controlled by user

Reply:
If GcsAuditLogEnabled is true we check whether the user has set the header or not. If they have we give them priority, if they have not we set it ourselves.